### ISSUE TITLE
New submission by Journal Manager does not appear in Unassigned

### ISSUE THREAD

**Issue by mfelczak** [pkp/pkp-lib#7270]
**Describe the bug**
When a Journal Manager creates a new submission there is no one assigned to the submission stage but the submission does not appear in the Unassigned tab.

**To Reproduce**
1. Have 1 Journal Manager whose role is assigned to 0 stages.
2. Have 0 Journal Editors or other user groups with Journal Manager permissions who are assigned to stages.
3. Submit as the Journal Manager.
4. Submit to a section that has 0 section editors assigned to it.

**What application are you using?**
OJS 3.3.0-7

**Additional information**
@NateWr requested the following from the OJS database.

```
mysql> select status from submissions where submission_id=17962;
+--------+
| status |
+--------+
|      1 |
+--------+
```

```
mysql> select sa.user_id,sa.user_group_id,ug.role_id from stage_assignments as sa left join user_groups as ug on (ug.user_group_id = sa.user_group_id) where sa.submission_id=17962;
+---------+---------------+---------+
| user_id | user_group_id | role_id |
+---------+---------------+---------+
|       1 |             2 |      16 |
+---------+---------------+---------+
```

**Comment by NateWr** [comment]
@mfelczak the result of the second sql query suggests that there is a user assigned to the submission.

**Comment by NateWr** [comment]
> In My Queue there is a prompt that no editor has been assigned.

Ah, ok. So the problem is that there is an unassigned notice when there shouldn't be, right?

**Comment by mfelczak** [comment]
There are no participants that are visible in the Participants grid and the message "Assign an editor to enable the editorial decisions for this stage" is displayed on both the submission page as well as in the My Assigned and All Active lists (screenshots below).

![participants](https://user-images.githubusercontent.com/200793/132590330-4a267d1b-993b-4ffe-a4a7-2cbf97753785.png)

![my_assigned](https://user-images.githubusercontent.com/200793/132590351-451c40b3-fbbc-4eef-bad0-59f4d8b704dc.png)

![all_active](https://user-images.githubusercontent.com/200793/132590364-0e2963c8-bab2-4540-a660-474fb500d355.png)

**Comment by NateWr** [comment]
What are the role settings for that Journal Manager role?

**Comment by mfelczak** [comment]
Thanks Nate, confirming that no stages are selected for the JM role.

![roles](https://user-images.githubusercontent.com/200793/132707759-5cf716d7-5366-4e1d-b2a6-00dd98f9c90e.png)

**Comment by NateWr** [comment]
Ok, I was able to reproduce this. The conditions necessary to reproduce the problem are the following:

1. Have at least 1 Journal Manager assigned to 0 stages
2. Have 0 Journal Editors.
3. Submit as the Journal Manager.
3. Submit to a section that has 0 section editors assigned to it.

This results in the one stage assignment of the JM. However, since the JM has 0 assigned stages, the stage assignment does not result in a participant on any particular stage.

@mfelczak I think you could get around this by assigning your JM to a stage. You can give them the Journal Editor role (if this exists) or something like it. (They're all the same permission level so the user group can be anything, as long as you can assign them to the submission stage.)

It's still a bug -- a JM with no stages shouldn't be getting a stage assignment unless their user group has the submission stage (see [code](https://github.com/pkp/pkp-lib/blob/main/classes/submission/form/PKPSubmissionSubmitStep4Form.inc.php#L74)). So it will take some more work to figure out what's going on there.

I'll update the original report with the reproduction steps.

**Comment by mfelczak** [comment]
Thanks Nate, but this user already has the Journal Editor role assigned and there are other Editors as well.

**Comment by NateWr** [comment]
In that case, I'm not able to reproduce this... :/ Can you try playing with a clean install to identify some reproduction steps?

**Comment by pmangahis** [comment]
@NateWr Tested this on the PKP School 3.3 install - here's the [screen recording](https://www.youtube.com/watch?v=29XdYSkgwqE) of that issue I noticed.

**Comment by NateWr** [comment]
Ok, I think I was going down the wrong rabbit hole. I thought this was related to editor assignment during step 4 of the submission wizard. But I suspect it's related to the author assignment code during step 1 of the submisssion. The author is assigned to the submission without checking whether their submitting user group is configured for the submission stage (or any stage).

https://github.com/pkp/pkp-lib/blob/main/classes/submission/form/PKPSubmissionSubmitStep1Form.inc.php#L449-L451

This results in the ghost stage assignment I saw [above](https://github.com/pkp/pkp-lib/issues/7270#issuecomment-916207385).

I think it should be pretty easy to check the user group's stages and only assign them if the user group has a submission stage. @mfelczak would PS be able to create and test this solution? If so, I can handle the PRs/tests.

One last thing: given the scenario above, this fix will result in no one assigned to the submission. Is that what is intended? Or does the user expect to have themselves assigned as a JE (not JM)? If so, we might need to think about a solution that changes what roles are offered for the "Submit As". If I follow the video correctly, the user has JM and JE roles, but is only offered the chance to submit as JM. If they were submitting as JE, this problem wouldn't arise, and after submission the user would be assigned to the submission as a JE.

**Comment by willinsky** [comment]
We're seeing a version of this bug among the journals in this installation: https://ojs.stanford.edu/ojs/index.php 

Describe the bug
When an Author creates a new submission, the submission does not appear in the Unassigned tab but appears in the All Active tab with no editor assigned and the warning "No editor has been assigned to this submission." 

To Reproduce
1. Submit a manuscript as an author.

What application are you using?
OJS 3.3.0-5

**Comment by asmecher** [comment]
I think the several issues here were caused by the working-as-intended, but often confusing, implementation of automatic role assignments that shipped in 3.3.0 and prior. When only a single user was active in a role, OJS automatically assigned that user, and if this occurred with an Editor or Journal Manager role it would result in the submission jumping right over the Unassigned stage.

This has already been improved in 3.4.0.

I'll close this as a WONTFIX for 3.3.0 -- but if someone thinks I've misunderstood the issue, please add details.
